docs/DeveloperGuide.md

```diff
@@ -1,34 +1,579 @@
+
+**Step 3.** The `LogicManager` calls `DeleteCommand#execute`. The `TaskList` now removed task with index 1. The message to show if the task is deleted successfully is subsequently outputted by the `AthenaUi` class to the user.
+
+**Step 4.** After the command is executed, `LogicManager` calls `Storage#saveTaskListData` to automatically save the tasks in the `TaskList` into the save file.
+
+**Step 4.** Upon completion of execution, `LogicManager` returns a boolean value `false` to `Athena` to allow the continuous run of the program. 
+
+The following sequence diagram illustrates how Step 3 of the task deleting operation works:
+
+![DeleteTaskSequenceDiagram](sequenceDiagrams/DeleteCommand%20diagram.png)
+
+### Mark task as done feature
+The marking task as done mechanism is facilitated by `LogicManager`.
+
+The retrieving of input details is done by `Parser`. It splits the user input based on the command type and the index of task.
+The `DoneCommand` class will then be executed and the task will be marked as done in the `TaskList`.
+
+`Parser`, `DoneCommand`, `TaskList` implements the following operations:
+
+* `Parser#parse` - Parse user input to retrieve the command type and index of task.
+* `Parser#parseDoneCommand` - Parse user input to retrieve the index of task to be marked as done and creates an `DoneCommand` object.
+* `DoneCommand#execute` - Marks task as done into `TaskList` and calls `AthenaUi` to print message output.
+* `Storage#saveTaskListData` - Writes the current task list into the save file.
+
+Given below is an example usage scenario and how the marking task as done mechanism behaves at each step.
+
+**Step 1.** The user marks a task as done in the application, by inputting `done 1`. 
+
+**Step 2.** The input will be read in by the `Athena` class. The input will be passed into `LogicManager` where `Parser#parse` will parse the user input to get the command type and details then call`Parse#parseDoneCommand`to create a `DoneCommand` object. The `DoneCommand` object is returned to the `LogicManager`.
+
+**Step 3.** The `LogicManager` calls `DoneCommand#execute`. The `TaskList` now has task with index 1 marked as done. The message to show if the task is marked as done successfully is subsequently outputted by the `AthenaUi` class to the user.
+ 
+**Step 4.** After the command is executed, `LogicManager` calls `Storage#saveTaskListData` to automatically save the tasks in the `TaskList` into the save file.
+
+**Step 5.** Upon completion of execution, `LogicManager` returns a boolean value `false` to `Athena` to allow the continuous run of the program. 
+
+The following sequence diagram illustrates how Step 3 of the marking task as done operation works:
+
+![DoneTaskSequenceDiagram](sequenceDiagrams/DoneCommand%20diagram.png)
```

Comment:
Same as previous, the position of the return line for the DoneCommand and the naming of the header box and the removal of the footbox. Should the method name getTaskFromNumber() be aligned with the arrow? It looks a little confusing for now.
<img width="520" alt="Screenshot 2020-10-30 at 14 30 17" src="https://user-images.githubusercontent.com/56800018/97668464-fc0fcd80-1abc-11eb-95be-6973f086bf49.png">